### ISSUE TITLE
[QUESTION] Thread safety

### ISSUE THREAD

**Issue by iProgramMC** [lidgren/lidgren-network-gen3#141]
I wonder, is Lidgren safe to use in multiple threads? I plan on having multiple hosts on my program, and I'm wondering if its possible with multiple threads. Also, is it possible that multiple threads can access the same host? Do I need to add something more to make it thread-safe, or is it already thread-safe?

**Comment by guerrillacontra** [comment]
It is all ready thread safe 👍

**Comment by iProgramMC** [comment]
Sounds good. Also, does lidgren support multiple hosts/servers per program?

**Comment by guerrillacontra** [comment]
I assume so as it's not a singleton, night have to experiment though

**Comment by iProgramMC** [comment]
ah, will do later

**Comment by PJB3005** [comment]
Yes, multiple `NetPeer`s works just fine in the same process (as long as you bind them to a different host:port pair, of course). I'm using it to implement dual stack IPv6 (because I didn't know what dual stack sockets were at the time...).
